pkg/gitqlite/display_test.go

```diff
@@ -0,0 +1,41 @@
+package gitqlite
+
+import (
+	"bytes"
+	"encoding/csv"
+	"strings"
+	"testing"
+)
+
+func TestDisplayCSV(t *testing.T) {
+	instance, err := New(fixtureRepoDir, &Options{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rows, err := instance.DB.Query("select * from commits limit 10")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var b bytes.Buffer
+	err = DisplayDB(rows, &b, "csv")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	r := csv.NewReader(strings.NewReader(b.String()))
+
+	records, err := r.ReadAll()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	lineCount := len(records)
+
+	if lineCount != 11 {
+		t.Fatalf("expected 11 lines of output, got: %d", lineCount)
+	}
+
+	// TODO perhaps test the actual content of the lines?
```

Comment:
couldn't this be somewhat simply implemented with r.ReadLine and the values of the output could be compared. If you used the getContents function to pull out the rows and string concatenated the rows of the 2d output array it should be pretty simple